scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -3116,21 +3115,39 @@ def _pass_thru(
     return res[-len(parent_spec):]
 
 
-def _raw_unit_addr(node: dtlib_Node) -> bytes:
+def _raw_unit_addr(node: dtlib_Node, parent: dtlib_Node) -> bytes:
     # _map_interrupt() helper. Returns the unit address (derived from 'reg' and
     # #address-cells) as a raw 'bytes'
 
+    iparent: Optional[dtlib_Node] = parent
+    addr_len: Optional[int] = None
+
+    while iparent is not None:
+        addr_len = _address_cells_self(iparent)
+        if addr_len is None:
+            # check the parent(s) of interrupt-controller(s) too
+            iparent = iparent.parent
+        else:
+            break
+
+    if addr_len is None:
+        addr_len =  2  # Default value per DT spec.
+
+    if addr_len == 0:
+        return b''
+
     if 'reg' not in node.props:
         _err(f"{node!r} lacks 'reg' property "
              "(needed for 'interrupt-map' unit address lookup)")
 
-    addr_len = 4*_address_cells(node)
-
-    if len(node.props['reg'].value) < addr_len:
-        _err(f"{node!r} has too short 'reg' property "
+    addr_len *= 4
+    prop_len = len(node.props['reg'].value)
+    if  prop_len < addr_len or prop_len % 4 != 0:
+        _err(f"{node!r} has too short or incorrectly defined 'reg' property "
              "(while doing 'interrupt-map' unit address lookup)")
 
-    return node.props['reg'].value[:addr_len]
+    # address is a big endian value
+    return node.props['reg'].value[-addr_len:]
```

Comment:
Is this correct? Taking `addr_len` from the end could mean taking size cells, or a different (address, size) pair.

I also see that this problem wouldn't be caught by tests, because the `interrupt-map-test` node has `#size-cells = <0>`.

Reply:
Ah, yes, thank you. I forgot about the size. Will fix it.